lua/nvim-treesitter/ts_utils.lua

```diff
@@ -212,4 +212,52 @@ function M.get_node_at_cursor(winnr)
   return root:named_descendant_for_range(cursor[1]-1,cursor[2],cursor[1]-1,cursor[2])
 end
 
+
+-- Set visual selection to node
+function M.update_selection(buf, node)
```

Comment:
Please check whether this is correct and also works with `incremental  selection`

Reply:
It seems that `incremental_selection` has problems to increment some selections without the last exclusive character.

Reply:
I had this issue before that's why i added one columns to the selection each time. If you have some time, look into it, or i'll do it.